library-udf/src/test/java/org/apache/iotdb/library/dmatch/DMatchTests.java

```diff
@@ -0,0 +1,547 @@
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d1(timestamp,s2,s4) values(%d,%d,%d)", 1000, 100, 106));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s2,s4) values(%d,%d,%d)", 100, 100, 101));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s2,s4) values(%d,%d,%d)", 200, 102, 101));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s2,s4) values(%d,%d,%d)", 300, 104, 102));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s2,s4) values(%d,%d,%d)", 400, 126, 102));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s2,s4) values(%d,%d,%d)", 500, 108, 103));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s2,s4) values(%d,%d,%d)", 600, 112, 104));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s2,s4) values(%d,%d,%d)", 700, 114, 104));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s2,s4) values(%d,%d,%d)", 800, 112, 104));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s2,s4) values(%d,%d,%d)", 900, 118, 105));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s2,s4) values(%d,%d,%d)", 1000, 100, 106));
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  private static void registerUDF() {
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute(
+          "create function completeness as 'org.apache.iotdb.library.dmatch.UDAFCov'");
+      statement.execute("create function timeliness as 'org.apache.iotdb.library.dmatch.UDAFDtw'");
+      statement.execute(
+          "create function consistency as 'org.apache.iotdb.library.dmatch.UDAFPearson'");
+      statement.execute("create function validity as 'org.apache.iotdb.library.dmatch.UDTFPtnSym'");
+      statement.execute("create function validity as 'org.apache.iotdb.library.dmatch.UDTFXcorr'");
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    EnvFactory.getEnv().cleanAfterClass();
+    ConfigFactory.getConfig()
+        .setUdfCollectorMemoryBudgetInMB(100)
+        .setUdfTransformerMemoryBudgetInMB(100)
+        .setUdfReaderMemoryBudgetInMB(100);
+  }
+
+  @Test
+  public void testCov1() {
+    String sqlStr = "select cov(d1.s1,d1.s3) from root.vehicle";
```

Comment:
As we have discussed previously, please check the equalivent between expected and acutal data returned, not only row counts.